### PR TITLE
Add provider equinix and module_name in Equinix Metal TF configs

### PR DESCRIPTION
**Pull request by codinja1188**

### CODE REVIEW

File: examples/fabric-port-connection/versions.tf [hunk 1]
```diff
@@ -4,11 +4,14 @@ terraform {
...
+      version = "~> 1.14"
     }
     alicloud = {
       source  = "aliyun/alicloud"
       version = ">= 1.165.0"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection/fabric-port-connection"
+  }
```
Comment (ocobles) [inline examples/fabric-port-connection/versions.tf]:
```suggestion
```

File: examples/network-edge-device-connection/versions.tf [hunk 2]
```diff
@@ -4,11 +4,14 @@ terraform {
...
+      version = "~> 1.14"
     }
     alicloud = {
       source  = "aliyun/alicloud"
       version = ">= 1.165.0"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection-alibaba/network-edge-device-connection"
+  }
```
Comment (ocobles) [inline examples/network-edge-device-connection/versions.tf]: Examples don't need to define provider_meta, it's just for the module itself
```suggestion
```

File: examples/service-token-metal-to-alibaba-connection/versions.tf [hunk 3]
```diff
@@ -4,11 +4,14 @@ terraform {
...
+      version = "~> 1.14"
     }
     alicloud = {
       source  = "aliyun/alicloud"
       version = ">= 1.165.0"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection-alibaba/service-token-metal-to-alibaba-connection"
+  }
```
Comment (ocobles) [inline examples/service-token-metal-to-alibaba-connection/versions.tf]:
```suggestion
```

**Review by ocobles — CHANGES_REQUESTED** [review]
Please, review requested changes

**Comment by codinja1188** [comment]
> Please, review requested changes
Updated

File: versions.tf [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
-      version = ">= 1.7.0"
+      version = "~> 1.14"
     }
     alicloud = {
       source  = "aliyun/alicloud"
       version = ">= 1.165.0"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection-alibab"
```
Comment (ocobles) [inline versions.tf]:
```suggestion
    module_name = "equinix-fabric-connection-alibaba"
```

Comment (codinja1188) [reply versions.tf]: updated typo

**Review by ocobles — DISMISSED** [review]

File: main.tf [hunk 5]
```diff
@@ -9,20 +9,20 @@ resource "random_string" "this" {
 }
 
 module "equinix-fabric-connection" {
-  source = "github.com/equinix-labs/terraform-equinix-fabric-connection"
+  source  = "github.com/equinix-labs/terraform-equinix-fabric-connection"
```
Comment (ocobles) [inline main.tf]: Test complains on that source format, it should be:

```suggestion
  source  = "equinix-labs/fabric-connection/equinix"
```

Comment (ocobles) [reply main.tf]: Please update also `version = "0.3.1"` to `version = "0.4.0"`

Comment (codinja1188) [reply main.tf]: updated

**Review by displague — APPROVED** [review]

**Pull request merged**